python/src/main/java/org/apache/apex/malhar/python/operator/PythonGenericOperator.java

```diff
@@ -0,0 +1,269 @@
+
+import com.datatorrent.api.Context.DAGContext;
+import com.datatorrent.api.Context.OperatorContext;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.common.util.BaseOperator;
+
+import py4j.GatewayConnection;
+import py4j.GatewayServer;
+import py4j.GatewayServerListener;
+import py4j.Py4JServerConnection;
+
+public abstract class PythonGenericOperator<T> extends BaseOperator
+{
+
+  protected transient GatewayServer server = null;
+  protected transient PythonGatewayServerListenser py4jListener = null;
+  protected transient PythonWorkerProxy<T> pythonWorkerProxy = null;
+  protected byte[] serializedFunction = null;
+  private static final Logger LOG = LoggerFactory.getLogger(PythonGenericOperator.class);
+  protected transient OpType operationType = null;
+  private PythonWorkerContext pythonWorkerContext = null;
+  protected Map<String, String> environementData = new HashMap<String, String>();
+
+
+  public enum OpType
+  {
+    MAP("MAP"),
+    FLAT_MAP("FLAT_MAP"),
+    FILTER("FILTER");
+
+    private String operationName = null;
+
+    OpType(String name)
+    {
+      this.operationName = name;
+    }
+
+    public String getType()
+    {
+      return operationName;
+    }
+
+  }
+
+  public final transient DefaultInputPort<T> in = new DefaultInputPort<T>()
+  {
+    @Override
+    public void process(T tuple)
+    {
+
+      processTuple(tuple);
+
+    }
+
+  };
+  public final transient DefaultOutputPort<T> out = new DefaultOutputPort<T>();
+
+  public PythonGenericOperator()
+  {
+    this(null);
+
+  }
+
+  public PythonGenericOperator(byte[] serializedFunc)
+  {
+    this.serializedFunction = serializedFunc;
+    this.pythonWorkerContext = new PythonWorkerContext(this.operationType, serializedFunc, environementData);
+
+  }
+
+//  public PythonGenericOperator(PythonWorkerContext context)
+//  {
+//    this.pythonWorkerContext = context;
+//  }
+
+  public void setup(OperatorContext context)
+  {
+    LOG.debug("APPLICATION PATH FROM PYTHON OPERATOR" + (String)context.getValue(DAGContext.APPLICATION_PATH));
+    // Setting up context path explicitly for handling local as well Hadoop Based Application Development
+    this.pythonWorkerContext.setup();
+
+    this.pythonWorkerProxy = new PythonWorkerProxy<>(this.serializedFunction);
+
+    int port = NetworkUtils.findAvaliablePort();
+    // Instantiating Py4j Gateway Server for Python Worker Process connect back
+    this.server = new GatewayServer(this.pythonWorkerProxy, port);
+    this.py4jListener = new PythonGenericOperator.PythonGatewayServerListenser(this.server, this.pythonWorkerContext);
+    this.server.addListener(this.py4jListener);
+    this.server.start(true);
+
+    int pythonServerStartAttempts = 5;
+    while (!this.py4jListener.isPythonServerStarted() && !this.pythonWorkerProxy.isFunctionEnabled() && pythonServerStartAttempts > 0) {
+      try {
+        Thread.sleep(5000L);
```

Comment:
5 second is too big sleep. may be you could use notify to this thread to interrupt the sleep.